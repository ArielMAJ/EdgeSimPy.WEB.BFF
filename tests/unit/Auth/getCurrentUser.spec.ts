// import { createTestClient } from "apollo-server-testing";
// import { getCurrentUserQuery } from "./__gql__/queries";
// import { createTestServer } from "../../utils";
// import * as userService from "../../../src/services/authentication/user.service";

// beforeAll(async () => {});

// beforeEach(async () => {});

// afterAll(async () => {});

// describe("Get Current User", () => {
//   it("should return object with registered user", async () => {
//     jest.spyOn(userService, "getCurrentUser").mockResolvedValue({
//       id: 1,
//       name: "John Doe",
//       email: "user@example.com",
//       createdAt: new Date("2025-05-25T14:28:15.785Z"),
//       updatedAt: new Date("2025-05-25T14:28:15.785Z"),
//     });

//     const apolloServer = await createTestServer({
//       req: {
//         headers: {
//           authorization: "Bearer valid-token",
//         },
//       },
//     });
//     const { query } = createTestClient(apolloServer);

//     const res = await query({
//       query: getCurrentUserQuery,
//     });
//     expect(res).toMatchSnapshot("Example valid current user");
//   });

//   it("return null if no valid user is found", async () => {
//     jest.spyOn(userService, "getCurrentUser").mockResolvedValue(null);
//     const apolloServer = await createTestServer({
//       req: {
//         headers: {
//           authorization: "Bearer valid-token",
//         },
//       },
//     });
//     const { query } = createTestClient(apolloServer);

//     const res = await query({
//       query: getCurrentUserQuery,
//     });
//     expect(res).toMatchSnapshot("Example invalid current user");
//   });
// });
