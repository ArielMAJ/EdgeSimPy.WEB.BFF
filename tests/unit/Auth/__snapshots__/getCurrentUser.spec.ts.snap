// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Current User return null if no valid user is found: Example invalid current user 1`] = `
Object {
  "data": null,
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Get Current User return null if no valid user is found: Example valid current user 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [ValidationError: Cannot query field "currentUser" on type "Query".],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Get Current User should return object with registered user: Example valid current user 1`] = `
Object {
  "data": Object {
    "currentUser": Object {
      "id": "1",
      "name": "John Doe",
      "email": "user@example.com",
      "createdAt": "2025-05-25T14:28:15.785Z",
      "updatedAt": "2025-05-25T14:28:15.785Z"
    }
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
